### PR TITLE
[test/integration] Reuse envtest

### LINKED ISSUE #767: Job/MPIJob integration suite recreates the envtest for every test
**What would you like to be cleaned**:

We should recreate the kueue-manager, but not the envtest.

This requires some refactoring in the integration test framework.

Ex: https://github.com/kubernetes-sigs/kueue/blob/main/test/integration/controller/mpijob/mpijob_controller_test.go#L70
And https://github.com/kubernetes-sigs/kueue/blob/main/test/integration/controller/job/job_controller_test.go#L71

**Why is this needed**:

To speed up the run time of the suite

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Reuse envtest in the integration tests for batch,mpi, and ray jobs.

Doing so we speed up the overall integration tests with @35%.

#### Which issue(s) this PR fixes:

Fixes #767

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/integration/controller/job/job_controller_test.go [hunk 1]
```diff
@@ -429,17 +449,40 @@ var _ = ginkgo.Describe("Job controller", func() {
 	})
 })
 
-var _ = ginkgo.Describe("Job controller for workloads with no queue set", func() {
-	ginkgo.BeforeEach(func() {
+var _ = ginkgo.Describe("Job controller for workloads with no queue set", ginkgo.Ordered, func() {
```
Comment (alculquicondor) [inline test/integration/controller/job/job_controller_test.go:452]: why not add `ContinueOnFailure` as well?

File: test/integration/controller/job/job_controller_test.go [hunk 2]
```diff
@@ -46,31 +46,59 @@ import (
...
 		fwk.Teardown()
 	})
+
+	var (
+		jobNamespace      string
+		ns                *corev1.Namespace
+		wlLookupKey       types.NamespacedName
+		parentWlLookupKey types.NamespacedName
+		childWlLookupKey  types.NamespacedName
+		childLookupKey    types.NamespacedName
```
Comment (alculquicondor) [inline test/integration/controller/job/job_controller_test.go:81]: why move all of these variables to this level, if they are only used in one case?

File: test/integration/controller/job/job_controller_test.go [hunk 3]
```diff
@@ -46,31 +46,59 @@ import (
...
+	)
+
+	ginkgo.BeforeEach(func() {
+		ns := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "core-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+		jobNamespace = ns.Name
```
Comment (alculquicondor) [inline test/integration/controller/job/job_controller_test.go:91]: get rid of this variable. Just use `ns.Name` were needed.

File: test/integration/controller/job/job_controller_test.go [hunk 4]
```diff
@@ -429,17 +449,40 @@ var _ = ginkgo.Describe("Job controller", func() {
...
-	ginkgo.AfterEach(func() {
+	ginkgo.AfterAll(func() {
 		fwk.Teardown()
 	})
+
+	var (
+		jobNamespace   string
+		ns             *corev1.Namespace
+		wlLookupKey    types.NamespacedName
+		childLookupKey types.NamespacedName
```
Comment (alculquicondor) [inline test/integration/controller/job/job_controller_test.go:468]: only create were needed

File: test/integration/controller/rayjob/rayjob_controller_test.go [hunk 5]
```diff
@@ -79,10 +76,30 @@ var _ = ginkgo.Describe("Job controller", func() {
...
 		ctx, cfg, k8sClient = fwk.Setup()
 	})
-	ginkgo.AfterEach(func() {
+	ginkgo.AfterAll(func() {
 		fwk.Teardown()
 	})
 
+	var (
+		ns           *corev1.Namespace
+		jobNamespace string
```
Comment (alculquicondor) [inline test/integration/controller/rayjob/rayjob_controller_test.go:85]: use `ns.Name`

**Review by tenzen-y — COMMENTED** [review]
Thanks! Great improvement!
/lgtm

**Comment by tenzen-y** [comment]
Probably, the kube-apiserver hangs up :(

```shell
{Unexpected error:
    <*fmt.wrapError | 0xc0001264a0>: 
    unable to install CRDs onto control plane: something went wrong waiting for CRDs to appear as API resources: timed out waiting for the condition
    {
        msg: "unable to install CRDs onto control plane: something went wrong waiting for CRDs to appear as API resources: timed out waiting for the condition",
        err: <*fmt.wrapError | 0xc000126480>{
            msg: "something went wrong waiting for CRDs to appear as API resources: timed out waiting for the condition",
            err: <*errors.errorString | 0xc00037f0d0>{
                s: "timed out waiting for the condition",
            },
        },
    }
occurred failed [FAILED] Unexpected error:
    <*fmt.wrapError | 0xc0001264a0>: 
    unable to install CRDs onto control plane: something went wrong waiting for CRDs to appear as API resources: timed out waiting for the condition
    {
        msg: "unable to install CRDs onto control plane: something went wrong waiting for CRDs to appear as API resources: timed out waiting for the condition",
        err: <*fmt.wrapError | 0xc000126480>{
            msg: "something went wrong waiting for CRDs to appear as API resources: timed out waiting for the condition",
            err: <*errors.errorString | 0xc00037f0d0>{
                s: "timed out waiting for the condition",
            },
        },
    }
occurred
In [BeforeSuite] at: /home/prow/go/src/sigs.k8s.io/kueue/test/integration/controller/core/suite_test.go:61 @ 05/31/23 07:55:40.17
}
```

**Comment by tenzen-y** [comment]
The test is killed again...

```
/home/prow/go/src/sigs.k8s.io/kueue/bin/golangci-lint run --timeout 7m0s
make: *** [Makefile:162: ci-lint] Killed
```

**Comment by alculquicondor** [comment]
/retest

after #823

**Pull request merged**